File: global/global-workflow-boot-starter/src/main/java/com/particle/global/workflow/dto/WorkflowNodeData.java
package com.particle.global.workflow.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.util.List;
import java.util.Map;

/**
 * 工作流节点数据 — 对应前端 node.data
 * <p>
 * 与前端 {@code WorkflowNodeData} 接口字段一一对应。
 * typeName/name 是前端 UI 展示用，后端不使用但保留（JSON 反序列化不丢失）。
 * </p>
 *
 * @author particle
 * @since 2026-05-03
 */
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class WorkflowNodeData {

    /** 节点类型名称（如"文本节点"），前端展示用 */
    private String typeName;

    /** 节点自定义名称，用户可编辑 */
    private String name;

    /** 节点角色: DATA / PROCESSOR */
    private String nodeRole;

    /** 节点执行类型: TEXT / IMAGE / VIDEO / HTTP / AI 等 */
    private String nodeType;

    /** 输入端口声明 */
    private Map<String,WorkflowNodePort> inputPorts;

    /** 输出端口声明 */
    private Map<String,WorkflowNodePort> outputPorts;

    /** 预设值端口（有值） */
    private Map<String,WorkflowNodePort> valuePorts;

    /** 节点行为配置（不含数据值） */
    private Map<String, Object> config;
}

File: web/global/pc/common/vueflowWorkflow/workflow/WorkflowNodeData.ts
import type { NodeRole, NodeType } from './enums'
import type { PortDefinition } from './PortDefinition'

/**
 * 节点运行时数据 (存在 node.data 中)
 *
 * 三种端口：
 * - inputPorts: 声明需要什么输入，值由引擎从上游注入
 * - outputPorts: 声明产生什么输出，值由 executor 执行后写入
 * - valuePorts: 节点自身携带的预设值，由用户在画布上填写
 *
 * config 只存行为配置（method、delay、operation 等）
 *
 * 与后端 WorkflowNodeDataDTO 对应
 */
export interface WorkflowNodeData {
    // ==================== UI 展示相关 ====================
    /** 节点类型名称（如"文本节点"），用于标识节点类型 */
    typeName: string

    /** 节点自定义名称/标题（如"系统提示词"），用户可编辑 */
    name: string

    // ==================== DAG 执行相关 ====================
    /** 节点角色: DATA / PROCESSOR */
    nodeRole: NodeRole

    /** 节点执行类型 */
    nodeType: NodeType

    /**
     * 节点行为配置（不含数据值，数据值走 valuePorts）
     *
     * PROCESSOR 节点 config 示例 (HTTP):
     *   { "method": "POST", "headers": {} }
     */
    config?: Record<string, any>

    // ==================== 端口定义（IO 声明）====================
    /** 输入端口声明（无值，引擎自动注入） */
    inputPorts?: Record<string, PortDefinition>

    /** 输出端口声明（无值，executor 执行后写入） */
    outputPorts?: Record<string, PortDefinition>

    /** 预设值端口（有值，用户在画布上填写） */
    valuePorts?: Record<string, PortDefinition>
}
